test/conformance/helpers/sources/source_crd_rbac_test_helper.go

```diff
@@ -0,0 +1,123 @@
+	"knative.dev/pkg/apis/duck"
+)
+
+var clusterRoleName = "eventing-sources-source-observer"
+var clusterRoleLabel = map[string]string{
+	duck.SourceDuckVersionLabel: "true",
+}
+
+func SourceCRDRBACTestHelperWithComponentsTestRunner(
+	t *testing.T,
+	sourceTestRunner testlib.ComponentsTestRunner,
+	options ...testlib.SetupClientOption,
+) {
+
+	sourceTestRunner.RunTests(t, testlib.FeatureBasic, func(st *testing.T, source metav1.TypeMeta) {
+		client := testlib.Setup(st, true, options...)
+		defer testlib.TearDown(client)
+
+		// From spec:
+		// Each source MUST have the following:
+		// kind: ClusterRole
+		// apiVersion: rbac.authorization.k8s.io/v1
+		// metadata:
+		//   name: foos-source-observer
+		//   labels:
+		//     duck.knative.dev/source: "true"
+		// rules:
+		//   - apiGroups:
+		//       - example.com
+		//     resources:
+		//       - foos
+		//     verbs:
+		//       - get
+		//       - list
+		//       - watch
+		st.Run("Source CRD has source observer cluster role", func(t *testing.T) {
+			ValidateRBAC(st, client, source)
+		})
+
+	})
+}
+
+func ValidateRBAC(st *testing.T, client *testlib.Client, object metav1.TypeMeta) {
+	labelSelector := &metav1.LabelSelector{
+		MatchLabels: clusterRoleLabel,
+	}
+
+	sourcePluralName := getSourcePluralName(client, object)
+
+	if !clusterRoleMeetsSpecs(client, labelSelector, sourcePluralName) {
+		//CRD Spec says new sources MUST include a ClusterRole as part of installing themselves into a cluster - so can't enforce it. Nothing to do here
+		//client.T.Fatalf("can't find source observer cluster role for CRD %q", object)
+		client.T.Logf("can't find source observer cluster role for CRD %q", object)
+	}
+}
+
+func getSourcePluralName(client *testlib.Client, object metav1.TypeMeta) string {
+	gvr, _ := meta.UnsafeGuessKindToResource(object.GroupVersionKind())
+	crdName := gvr.Resource + "." + gvr.Group
+
+	crd, err := client.Apiextensions.CustomResourceDefinitions().Get(context.Background(), crdName, metav1.GetOptions{
+		TypeMeta: metav1.TypeMeta{},
+	})
+	if err != nil {
+		client.T.Errorf("error while getting %q:%v", object, err)
+	}
+	return crd.Spec.Names.Plural
+}
+
+func clusterRoleMeetsSpecs(client *testlib.Client, labelSelector *metav1.LabelSelector, crdSourceName string) bool {
+	crs, err := client.Kube.RbacV1().ClusterRoles().List(context.Background(), metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", clusterRoleName).String(), //Cluster Role with name "eventing-sources-source-observer"
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),                         //Cluster Role with duck.knative.dev/source: "true" label
+	})
+	if err != nil {
+		client.T.Errorf("error while getting cluster roles %v", err)
+	}
+
+	for _, cr := range crs.Items {
+		for _, pr := range cr.Rules {
+			if contains(pr.Resources, crdSourceName) { //Cluster Role has the eventing source listed in Resources for a Policy Rule
+				return true
+			}
```

Comment:
Should we also check the verbs are what we expect them to be?

Reply:
Yes, sounds like that will be good. Added.